subsys/bluetooth/audio/tbs_client.c

```diff
@@ -600,6 +599,69 @@ static uint8_t notify_handler(struct bt_conn *conn,
 	return BT_GATT_ITER_CONTINUE;
 }
 
+/* Common function to read tbs_client strings which may require long reads */
+static uint8_t handle_string_long_read(struct bt_conn *conn, uint8_t err,
+				       struct bt_gatt_read_params *params, const void *data,
+				       uint16_t length, bt_tbs_client_read_string_cb cb)
+{
+	struct bt_tbs_instance *inst = CONTAINER_OF(params, struct bt_tbs_instance, read_params);
+	uint16_t offset = params->single.offset;
+	uint8_t inst_index = tbs_index(conn, inst);
+	const char *received_string;
+	int tbs_err = err;
+
+	if ((tbs_err == 0) && (data != NULL) &&
+	    (net_buf_simple_tailroom(&inst->net_buf) < length)) {
+		tbs_err = BT_ATT_ERR_INSUFFICIENT_RESOURCES;
+	}
```

Comment:
I suppose that there are some string values that can be truncated, instead of just providing an error, right? Something like the Friendly name can trivially be truncated, and perhaps even something like the URI list can as well, if it is correctly truncated by truncating an entire URI (and not stopping in the middle of an URI).

Reply:
That's right, I'll change the function to be able to truncate without complaining.